### DISCUSSION TITLE
Will it work for Webdriver 3.x & Chrome v 96+

### DISCUSSION THREAD

**Discussion by madhavchoudhary83** [sukgu/shadow-automation-selenium#53] [Q&A]
Will it work for Webdriver 3.x & Chrome v 96+, or we need to upgrade webdriver to 4.x to utilize this solution ? I red that there are fixes made in webdriver 4.x to get  the shadow root in Chrome 96 and above.

I tried to use the solution but its unable to get webelement for the parent (shadowhost) as well as elements under the parent.
With core webdriver findelement method, the same shadow host is getting detected.

![image](https://user-images.githubusercontent.com/66959592/179670603-5fd06c4a-2c7a-4b55-9164-676ea654b0d0.png)

`          WebElement parent1 = this.driver.findElement(By.cssSelector("#search")); 
          System.out.println("XXXXXXXXXXXXXXXXXXXXXXXXX "+ parent1.isDisplayed());
           Shadow shadow = new Shadow( this.driver);             
                WebElement parent = shadow.findElementByXPath("//*[@id='search']");
           //    WebElement parent = shadow.findElement("#search");
          //    WebElement searchTextbox = shadow.findElement("input[placeholder='Search']");
                WebElement searchTextbox = shadow.findElement(parent, "input[placeholder='Search']");
`

In above code snippet, line 1 works however, line 4 not working also , tried with line  6 & 7 and that also fails at 6.
Failing as NoSuchElementException

**Comment by sukgu** [comment]
@madhavchoudhary83 
Can you please try this in code?
```WebElement searchTextbox = shadow.findElement("coral-search-field#search input[placeholder='Search']");```

And this in your browser console?
```document.querySelector('coral-search-field#search').shadowRoot.querySelector('input[placeholder='Search']')```

If the 2nd output is blank then we need to understand the DOM first to find a unique way to get Element. Else you can debug it and raise a bug with your DOM snapshot.

**Reply by madhavchoudhary83** [reply]
Its working , the issue was that earlier the driver instance i was using was from EventFiringWebDriver instead of WebDriver, After realizing this issue, fixed the driver instance and it worked.

Thanks.
